server/src/main/kotlin/suwayomi/tachidesk/manga/impl/MangaList.kt

```diff
@@ -73,7 +76,42 @@ object MangaList {
             // delete thumbnail in case cached data still exists
             insertedMangaUrlsToId.forEach { (_, id) -> Manga.clearThumbnail(id) }
 
-            val mangaUrlsToId = existingMangaUrlsToId + insertedMangaUrlsToId
+            val mangaToUpdate =
+                mangas.mapNotNull { sManga ->
+                    existingMangaUrlsToId[sManga.url]?.let { sManga to it }
+                }.filterNot { (_, resultRow) ->
+                    resultRow[MangaTable.inLibrary]
+                }
+
+            mangaToUpdate.forEach { (sManga, manga) ->
+                if (manga[MangaTable.thumbnail_url] != sManga.thumbnail_url) {
+                    Manga.clearThumbnail(manga[MangaTable.id].value)
+                }
+            }
+
+            BatchUpdateStatement(MangaTable).apply {
+                mangaToUpdate.forEach { (sManga, manga) ->
+                    addBatch(EntityID(manga[MangaTable.id].value, MangaTable))
+                    this[MangaTable.title] = sManga.title
+                    this[MangaTable.artist] = sManga.artist
+                    this[MangaTable.author] = sManga.author
+                    this[MangaTable.description] = sManga.description
+                    this[MangaTable.genre] = sManga.genre
+                    this[MangaTable.status] = sManga.status
+                    this[MangaTable.thumbnail_url] = sManga.thumbnail_url
+                    this[MangaTable.updateStrategy] = sManga.update_strategy.name
+                    if (!sManga.thumbnail_url.isNullOrEmpty() && manga[MangaTable.thumbnail_url] != sManga.thumbnail_url) {
+                        this[MangaTable.thumbnailUrlLastFetched] = Instant.now().epochSecond
```

Comment:
could `Manga.clearThumbnail` be called from here?

Reply:
Probably